more_itertools/more.py

```diff
@@ -1873,9 +1875,12 @@ def divide(n, iterable):
         >>> [list(c) for c in children]
         [[1], [2], [3], [], []]
 
-    This function will exhaust the iterable before returning and may require
-    significant storage. If order is not important, see :func:`distribute`,
-    which does not first pull the iterable into memory.
+    This function will convert the iterable to a tuple upfront and thus may
```

Comment:
That's not true. It only does that if the slicing attempt fails.

Reply:
Fair. Would it be better to revert the phrasing here back to "exhaust the iterable" or would inserting "(if needed)" suffice?

Reply:
Not sure. It's tricky. "Exhausting" a list for example even sounds *wrong*. And the later slices `seq[start:stop]` can *also* use significant storage.